### PR TITLE
Registering Git Profile for Windows Terminal on Git.Git manifests

### LINKED ISSUE #109915: [Package Issue]: Git.Git - enable Terminal integration
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

Windows Terminal no longer autodetects Git Bash and now expects Git to register itself (which is safe to do even if Terminal is not installed): https://github.com/microsoft/terminal/issues/15559

Unfortunately the installer option is off by default and upstream will not change it: https://github.com/git-for-windows/git/issues/4468

Can you add `/enable-component windowsterminal` to install flags?
Not sure if that's the right syntax but that's how the option is named in https://github.com/git-for-windows/build-extra/blob/main/installer/install.iss#L115

### Steps to reproduce

```
winget install -e --id Microsoft.WindowsTerminal
winget install -e --id Git.Git
wt
```

### Actual behavior

No git bash profile

### Expected behavior

Git Bash profile is on the list

### Environment

```raw
-
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by BrandonWanHuanSheng**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

Resolves https://github.com/microsoft/winget-pkgs/issues/109915

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/110011&drop=dogfoodAlpha

### CODE REVIEW

File: manifests/g/Git/Git/2.41.0/Git.Git.installer.yaml [hunk 1]
```diff
@@ -15,17 +15,23 @@ Commands:
...
+  Custom: /SUPPRESSMSGBOXES /enable-component windowsterminal
 Installers:
 - Architecture: x64
   Scope: machine
   InstallerUrl: https://github.com/git-for-windows/git/releases/download/v2.41.0.windows.1/Git-2.41.0-64-bit.exe
   InstallerSha256: 45dc30410916b8ec5501be39d01d5b60535731c04fa68283b4f9df4920877d4e
   ElevationRequirement: elevatesSelf
+  Dependencies:
+    PackageDependencies:
+    - PackageIdentifier: Microsoft.WindowsTerminal
```
Comment (mwisnicki) [inline manifests/g/Git/Git/2.41.0/Git.Git.installer.yaml:27]: This is incorrect. `/enable-component windowsterminal` does not introduce dependency on Windows Terminal.
The registration is safe to do even if WT is not installed: https://github.com/microsoft/terminal/issues/15559#issuecomment-1593684739

**Comment by quhxl** [comment]
I'm honestly not sure if winget, as a package manager, should have any influence on the default installer settings. Shouldn't we leave it up to the developers to decide which features they see as relevant to most people, rather than influencing that sort of things?

**Comment by BrandonWanHuanSheng** [comment]
> I'm honestly not sure if winget, as a package manager, should have any influence on the default installer settings. Shouldn't we leave it up to the developers to decide which features they see as relevant to most people, rather than influencing that sort of things?

Yes.

**Comment by BrandonWanHuanSheng** [comment]
Stephengillie, It look like wingetbot give me Internal-Error-Dynamic-Scan

**Comment by BrandonWanHuanSheng** [comment]
It look like the github bots keep mentioning me. It is so loud.

**Comment by BrandonWanHuanSheng** [comment]
It look like the pipelines validation was suspended.

**Comment by BrandonWanHuanSheng** [comment]
Restarting Azure Pipelines.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**